ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelParameterizedTest.java

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.router.AsyncWritableChannel;
+import com.github.ambry.router.Callback;
+import com.github.ambry.router.FutureResult;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests functionality of {@link ByteBufferReadableStreamChannel}.
+ */
+@RunWith(Parameterized.class)
+public class ByteBufferReadableStreamChannelParameterizedTest {
+  public static final int BUFFER_SIZE = 256;
+
+  private ByteBuffer contentWrapper;
+  private ByteBufferReadableStreamChannel byteBufferReadableStreamChannel;
+
+  public ByteBufferReadableStreamChannelParameterizedTest(List<ByteBuffer> bufferList, ByteBuffer content1,
```

Comment:
instead of passing the verifying buffers as parameters, you could just create a copy of what is in the list, no?